Hunting Queries/MultipleDataSources/RareDNSLookupWithDataTransfer.txt

```diff
@@ -0,0 +1,96 @@
+| summarize LookupStartTime = min(TimeGenerated), LookupEndTime = max(TimeGenerated), LookupCountToday = count() by ClientIP, Domain, IPAddresses 
+| project LookupStartTime, LookupEndTime, ClientIP, LookupCountToday, Domain, IPAddresses;
+// Remove Common Lookups from lookback period from Todays lookups
+let UncommonLookupsToday = TodayLookups
+| join kind=leftanti ( 
+CommonLookups
+)
+on Domain;
+// Join back the Daily Average Lookup Count to add context to rarity over lookback period
+let RareLookups = UncommonLookupsToday | join kind= innerunique (
+DomainLookups 
+) on Domain 
+| project LookupStartTime, LookupEndTime, ClientIP, Domain, IPAddresses, LookupCountToday, DailyAvgLookupCountOverLookback;
+let DNSIPBreakout = RareLookups
+| extend DnsIPAddress = iff(IPAddresses has ",", split(IPAddresses, ","), todynamic(IPAddresses)) 
+| mvexpand DnsIPAddress
+| extend DnsIPAddress = tostring(DnsIPAddress)
+| distinct LookupStartTime, LookupEndTime, ClientIP, Domain, DnsIPAddress, LookupCountToday, DailyAvgLookupCountOverLookback
+| extend DnsIPType = iff(DnsIPAddress matches regex PrivateIPregex,"private" ,"public" )
+| where DnsIPType =="public"
+;
+let DataMovement = ( union isfuzzy=true 
+(CommonSecurityLog
+| where TimeGenerated >= endtime
+| where DeviceVendor =="Palo Alto Networks" and Activity == "TRAFFIC"
+| extend DestinationIPType = iff(DestinationIP matches regex PrivateIPregex,"private" ,"public" )
+| where DestinationIPType =="public"
+| project DataType = DeviceVendor, TimeGenerated, SourceIP , SourcePort , DestinationIP, DestinationPort, ReceivedBytes, SentBytes
+| sort by SourceIP asc, SourcePort asc,TimeGenerated asc, DestinationIP asc, DestinationPort asc
+| summarize sum(ReceivedBytes), sum(SentBytes), count() by DataType, SourceIP , SourcePort , DestinationIP, DestinationPort
+| sort by sum_SentBytes desc
+),
+(WireData
+| where TimeGenerated >= endtime
+| where Direction == "Outbound"
+| extend RemoteIPType = iff(RemoteIP matches regex PrivateIPregex,"private" ,"public" ) 
+| where RemoteIPType =="public" 
+| project DataType = Type, TimeGenerated , SourceIP = LocalIP , SourcePort = LocalPortNumber , DestinationIP = RemoteIP, DestinationPort = RemotePortNumber, ReceivedBytes, SentBytes 
+| summarize sum(ReceivedBytes), sum(SentBytes), count() by DataType, SourceIP , SourcePort , DestinationIP, DestinationPort
+| extend DataType = Type
+| sort by sum_SentBytes desc
+),
+(VMConnection 
+| where TimeGenerated >= endtime
+| where Direction == "outbound"
+| extend DestinationIPType = iff(DestinationIp matches regex PrivateIPregex,"private" ,"public" )
+| where DestinationIPType =="public"
+| project DataType = Type, TimeGenerated , SourceIP = SourceIp , DestinationIP = DestinationIp, DestinationPort, ReceivedBytes = BytesReceived, SentBytes = BytesSent 
+| summarize sum(ReceivedBytes), sum(SentBytes), count() by DataType, SourceIP , DestinationIP, DestinationPort
```

Comment:
avoid implicit count() column name -- especially as this makes it through to the final output?

Reply:
Changed to ConnectionCount = count()